src/_http/index.js

```diff
@@ -123,17 +123,29 @@ function getDefaultHeaders() {
     driver: ['javascript', packageJson.version].join('-'),
   }
 
-  if (util.isNodeEnv()) {
-    driverEnv.runtime = ['nodejs', process.version].join('-')
-    driverEnv.env = util.getNodeRuntimeEnv()
-    var os = require('os')
-    driverEnv.os = [os.platform(), os.release()].join('-')
-  } else {
-    driverEnv.runtime = util.getBrowserDetails()
-    driverEnv.env = 'unknown'
-    driverEnv.os = getBrowserOsDetails()
+  let isServiceWorker
+
+  try {
+    isServiceWorker = global instanceof ServiceWorkerGlobalScope
+  } catch (error) {
+    isServiceWorker = false
   }
 
+  try {
+    if (util.isNodeEnv()) {
+      driverEnv.runtime = ['nodejs', process.version].join('-')
+      driverEnv.env = util.getNodeRuntimeEnv()
+      var os = require('os')
+      driverEnv.os = [os.platform(), os.release()].join('-')
+    } else if (isServiceWorker) {
+      driverEnv.runtime = 'Service Worker'
+    } else {
+      driverEnv.runtime = util.getBrowserDetails()
+      driverEnv.env = 'browser'
+      driverEnv.os = getBrowserOsDetails()
+    }
+  } catch (_) {}
```

Comment:
Same here. Is this syntax ECMA 5 compatible?

Reply:
yes. to be honest, the driver isn't es5 compatible. just an example, promises is es6

Reply:
It used to be a policy that JS driver was supposed be ECMA5 compatible. It had a pollyfill for promises, I believe. If this is not the case anymore, we should review the policy has changed or if we introduced a regression.

Reply:
driver doesn't have pollfyfill for promises. even more, doc said that it's es6 promises
https://github.com/fauna/faunadb-js#instantiating-a-client-and-issuing-queries

also, we are working on modularization which would have bundle tools from es6+ to anything, including es5.


https://caniuse.com/?search=es6
according to this table, only pretty old browsers doesn't support es6

Reply:
I don't disagree and I'm fine moving to es6. Let's just make sure this is done via a conscious and well documented decision rather than through a non-backwards compatible change.